### DISCUSSION TITLE
Homebrew not working behind proxy

### DISCUSSION THREAD

**Discussion by krey** [Homebrew/discussions#3056] [Getting started]
Hello,

Noob here, trying to use homebrew from behind a corporate proxy.

```
~ % brew install wget
==> Downloading https://ghcr.io/v2/homebrew/core/libunistring/manifests/1.0
curl: (22) The requested URL returned error: 407

Error: wget: Failed to download resource "libunistring_bottle_manifest"
Download failed: https://ghcr.io/v2/homebrew/core/libunistring/manifests/1.0
```

I'm trying to debug this error, so I curl the URL from a network where I don't need a proxy:

```
~$ curl -u me:my_token https://ghcr.io/v2/homebrew/core/libunistring/manifests/1.0
{"errors":[{"code":"DENIED","message":"invalid token"}]}
```

The token seems to work fine for other API requests, so I'd have thought it was valid...

**Comment by gromgit** [comment]
https://docs.brew.sh/Manpage#using-homebrew-behind-a-proxy

**Comment by SMillerDev** [comment]
That's really something you'll have to debug with the people who manage the proxy.

**Comment by carlocab** [comment]
`curl https://ghcr.io/v2/homebrew/core/libunistring/manifests/1.0` really isn't expected to work.

To even get anything meaningful out of it, you need to pass the headers 
```
Authorization: Bearer QQ==
```
and
```
Accept: application/vnd.oci.image.index.v1+json
```

**Reply by krey** [reply]
Thank you @carlocab, this is what I was looking for.

What I don't understand is, running
```
all_proxy="http://my-proxy:8080" brew install wget
```
still errors as above, complaining about proxy auth, but
```
all_proxy="http://my-proxy:8080" curl -H "Authorization: Bearer QQ=="\
-H "Accept: application/vnd.oci.image.index.v1+json"\
https://ghcr.io/v2/homebrew/core/wget/manifests/1.21.3
```
works fine.

Is homebrew doing something different under the hood?

I'm trying to reproduce the error using curl so I can submit to our proxy team.

**Reply by carlocab** [reply]
> Is homebrew doing something different under the hood?

I wouldn't call it different. `brew` downloads `https://ghcr.io/v2/homebrew/core/wget/manifests/1.21.3` to get the bottle manifest, and then uses the bottle manifest to download a bottle. So the command you've run isn't sufficient to download a bottle.
